src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -392,11 +392,19 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             vkey != VK_SNAPSHOT &&
             keyDown)
         {
+            if (modifiers.IsCtrlPressed() && vkey == 'A')
+            {
+                auto lock = _terminal->LockForWriting();
+                _terminal->SelectAll();
+                _renderer->TriggerSelection();

```

Comment:
I'm surprised we don't have a helper for this!

Reply:
yeah, I think at one point I made (or witnessed) the decision that triggering the renderer and (un)locking would be handled at the control core layer consistently to ensure we don't deadlock on accident. It's a little annoying, but we could swing the pendulum the other way hahaha